File: 55-jump-game.go
func canJump(nums []int) bool {
    furthest := 0
    for idx, n := range(nums){
        if furthest < idx{
            break
        }
        furthest = max(n+idx, furthest)
    }
    return furthest >= len(nums) - 1
}

func max(a, b int) int{
    if a > b{
        return a
    }
    return b
}

File: bwu/greedy/55-jump-game.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        furthest = 0
        for idx, val in enumerate(nums):
            if idx <= furthest and idx + val > furthest:
                furthest = idx + val
                
        return furthest >= len(nums) - 1
